src/Symfony/Component/Finder/Iterator/FileTypeFilterIterator.php

```diff
@@ -26,8 +26,8 @@ class FileTypeFilterIterator extends \FilterIterator
     private int $mode;
 
     /**
-     * @param \Iterator $iterator The Iterator to filter
-     * @param int       $mode     The mode (self::ONLY_FILES or self::ONLY_DIRECTORIES)
+     * @param \Iterator<string, \SplFileInfo> $iterator The Iterator to filter
```

Comment:
Should we also require our `SplFileInfo` implementation here?

Reply:
Well the question is what we want to achieve here. We have no real control over who is using these iterators besides the Finder implementation itself. Any changes we make here could be perceived as introducing new "bugs" by people who use the iterators in a legitimate/compatible way. Personally, I'm clearly aiming at bug fixes without such potentially controversial side-effects here.

More detailed explanation:

- RecursiveDirectoryIterator::current returns Symfony SplFileInfo, so that's its Iterator value type.
- RecursiveDirectoryIterator is the starting point base iterator in Finder::searchInDirectory, hence the whole Finder iterator chain iterates over Symfony SplFileInfo, too.

So from a strictly internal point of view, all of the iterators use the Symfony SplFileInfo.

To determine which type of SplFileInfo class each iterator *can* use we need to look at the current() method and its calls/usages. Basically there are three cases here:
1. If the current() method is implemented explicitly and returns the Symfony SplFileInfo, that's the Iterator value type. Example: RecursiveDirectoryIterator
2. Usages that access one of the relative path methods require a Symfony SplFileInfo so the corresponding iterator needs to provide that. Example: PathFilterIterator
3. Usages that only access methods of \SplFileInfo can work with any iterator that provides \SplFileInfo or any subclass of it. Example: FileTypeFilterIterator